node/commons/compatibility-layer.ts

```diff
@@ -27,28 +27,30 @@ export const convertBiggyProduct = async (
 ) => {
   const categories: string[] = product.categories
     ? product.categories.map((_: any, index: number) => {
-        const subArray = product.categories.slice(0, index)
-        return `/${subArray.join('/')}/`
-      })
+      const subArray = product.categories.slice(0, index)
+      return `/${subArray.join('/')}/`
+    })
     : []
 
 
   const categoriesIds: string[] = product.categoryIds
-  ? product.categoryIds.map((_: any, index: number) => {
+    ? product.categoryIds.map((_: any, index: number) => {
       const subArray = product.categoryIds.slice(0, index + 1)
       return `/${subArray.join('/')}/`
     }).reverse()
-  : []
+    : []
 
   const skus: SearchItem[] = (product.skus || []).map(
     convertSKU(product, indexingType, tradePolicy)
   )
 
+  product.productSpecifications = product.productSpecifications ?? []
+
   const allSpecifications = product.productSpecifications.concat(getSKUSpecifications(product))
```

Comment:
```suggestion
const allSpecifications = (product.productSpecifications ?? []).concat(getSKUSpecifications(product))
```

just a suggestion :)